core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -734,14 +742,28 @@ public void addColumn(Session session, TableHandle tableHandle, ColumnMetadata c
         CatalogHandle catalogHandle = tableHandle.getCatalogHandle();
         ConnectorMetadata metadata = getMetadataForWrite(session, catalogHandle);
         metadata.addColumn(session.toConnectorSession(catalogHandle), tableHandle.getConnectorHandle(), column);
+
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogHandle);
+        if (catalogMetadata.getSecurityManagement() != CONNECTOR) {
+            TableMetadata tableMetadata = getTableMetadata(session, tableHandle);
+            CatalogSchemaTableName sourceTableName = new CatalogSchemaTableName(catalogHandle.getCatalogName(), tableMetadata.getTable());
+            systemSecurityMetadata.columnCreated(session, sourceTableName, column.getName());
+        }
     }
 
     @Override
     public void dropColumn(Session session, TableHandle tableHandle, ColumnHandle column)
     {
         CatalogHandle catalogHandle = tableHandle.getCatalogHandle();
         ConnectorMetadata metadata = getMetadataForWrite(session, catalogHandle);
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogHandle);
         metadata.dropColumn(session.toConnectorSession(catalogHandle), tableHandle.getConnectorHandle(), column);
+        if (catalogMetadata.getSecurityManagement() != CONNECTOR) {
+            String columnName = getColumnMetadata(session, tableHandle, column).getName();
+            TableMetadata tableMetadata = getTableMetadata(session, tableHandle);
```

Comment:
@djsstarburst wonder what was the testing strategy for this change.

It seems it doesn't work for some connectors.
For example, with Hive connector, `metadata.dropColumn` sets the `SemiTransactionalHiveMetastore` into `EXCLUSIVE_OPERATION_BUFFERED` state and then `getTableMetadata` fails


```
io.trino.testing.QueryFailedException: Unsupported combination of operations in a single transaction

	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:122)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:490)
	at io.trino.testing.QueryAssertions.assertUpdate(QueryAssertions.java:71)
	at io.trino.testing.AbstractTestQueryFramework.assertUpdate(AbstractTestQueryFramework.java:365)
	at io.trino.testing.AbstractTestQueryFramework.assertUpdate(AbstractTestQueryFramework.java:360)
	at io.trino.testing.BaseConnectorTest.testAddAndDropColumnName(BaseConnectorTest.java:4308)
	at io.trino.testing.BaseConnectorTest.testAddAndDropColumnName(BaseConnectorTest.java:4286)
...

	Suppressed: java.lang.Exception: SQL: ALTER TABLE tcn_lowercaseh3yyp7xswo DROP COLUMN lowercase
		at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:493)
		... 29 more
Caused by: io.trino.spi.TrinoException: Unsupported combination of operations in a single transaction
	at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore.checkReadable(SemiTransactionalHiveMetastore.java:2357)
	at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore.getTable(SemiTransactionalHiveMetastore.java:244)
	at io.trino.plugin.hive.HiveMetadata.doGetTableMetadata(HiveMetadata.java:583)
	at io.trino.plugin.hive.HiveMetadata.getTableMetadata(HiveMetadata.java:569)
	at io.trino.plugin.hive.HiveMetadata.getTableMetadata(HiveMetadata.java:563)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.getTableMetadata(ClassLoaderSafeConnectorMetadata.java:263)
	at io.trino.plugin.objectstore.ObjectStoreMetadata.getTableMetadata(ObjectStoreMetadata.java:377)
	at io.trino.metadata.MetadataManager.getTableMetadata(MetadataManager.java:450)
	at io.trino.metadata.MetadataManager.dropColumn(MetadataManager.java:763)
	at io.trino.execution.DropColumnTask.execute(DropColumnTask.java:99)
```